### DISCUSSION TITLE
Intergration stops controling actuators

### DISCUSSION THREAD

**Discussion by ce3a** [SukramJ/aiohomematic#1233] [General]
### I agree to the following

- [X] I have read the [documentation](https://github.com/danielperna84/custom_homematic#custom_homematic)
- [X] I am aware of the latest [release notes](https://github.com/danielperna84/custom_homematic/releases)
- [X] The Backend (CCU/Homegear/...) is working as expected and connected devices are controllable by its own UI.
- [X] I am running the latest version of the custom_component (and Home Assistant)

### The problem

Since one of the latest homematicip_local or home assistant updates, I am observing the following issue with the homematicip
integration.

After restarting home assistant, the homematicip integration is working as expected for a few hours or days. Then, suddenly, I am no longer able to control the actuators (garage door, shutters, etc.). I can still see the states of HmIP devices in ha, but actuators are not triggered. Controlling the actuators from the CCU3 WebUI still works.

Restarting ha, resolves the issue. But after a few hours or days the same issue occurs again.

 

### What version of HomematicIP (local) has the issue?

1.42

### What was the last working version of HomematicIP (local)?

not sure

### What type of installation are you running?

Home Assistant OS

### What type of installation are you running for your homematic backend?

RaspberryMatic Standalone

### Which version of your homematic backend are you running?

3.71.12.20230826

### What hardware are you running for your system?

Pi4

### Which config details do you use

- [ ] Username and password
- [ ] TLS
- [ ] callback data ([see](https://github.com/danielperna84/custom_homematic#callback_host-and-callback_port))

### Which interfaces do you use?

- [ ] Homematic IP
- [ ] Homematic (Bidcos-RF)
- [ ] Groups (Heating-Group)
- [ ] BidCos-Wired (HM-Wired)

**Comment by SukramJ** [comment]
No logs. 
No integration diagnostics.

--> No support possible

**Comment by ce3a** [comment]
I will provide logs as soon as it occurs again.

Is this enough to enable logging:

```yaml
logger:
  default: warning
  logs:
    hahomematic.client: info
    hahomematic.entity: info
    hahomematic.xml_rpc_server: info
```

What else could help?

**Comment by SukramJ** [comment]
That should work:

```yaml
logger:
  default: warning
  logs:
    hahomematic: info
```

[info](url) level should contain relevant warnings and error.
Also the CCU logs could be helpful.

**Please add the integration diagnostics (Klick on the Integration, ...).**

I don't think this is a problem with the integration, but with your environment, but we'll see.

**Comment by ce3a** [comment]
Is this the integration diagnostics? [config_entry-homematicip_local.json.txt](https://github.com/danielperna84/hahomematic/files/12742788/config_entry-homematicip_local.json.txt)

**Comment by SukramJ** [comment]
Not related to your problem, but you could disable the Bidcos interface.

**Comment by SukramJ** [comment]
Which actors are not controllable? All or only a few?

**Comment by ce3a** [comment]
> Which actors are not controllable? All or only a few?

I think all of them. I have observed the issue with the following devices.

- HmIP-DRSI4
- HmIP-BROLL
- HmIP-MOD-HO
- HmIP-MOD-OC8

**Comment by ce3a** [comment]
> could disable the Bidcos interface.

I have tried to disable the interface by adding the following to my `configurations.yaml`:

```yaml
homematic:
  interfaces:
    bidos_rf_enabled: false
```

But, it seems I am doing something wrong. This leads to the following error:

>  Invalid config for [homematic]: expected a dictionary for dictionary value @ data['homematic']['interfaces']['bidos_rf_enabled']. Got False. (See /config/configuration.yaml, line 56).  

Can you help?

**Comment by SukramJ** [comment]
Now I'm curious.

How do you get the idea to enter something in the configuration.yaml if you have previously carried out the configuration in the UI?

Simply click on the integration and configuration!

**Comment by ce3a** [comment]
You mean this?

![image](https://github.com/danielperna84/hahomematic/assets/5941722/e803c88f-125d-4f53-81a9-62b3ca7f6dbe)

I cannot find an option for disabling the Bidcos interface here.

**Comment by SukramJ** [comment]
Yes, and if you press on submit you'll get again the second page with further options.

**Comment by ce3a** [comment]
Ah, got it. Thanks.

**Comment by SukramJ** [comment]
Any news here? Otherwise this issue could be closed.

**Comment by ce3a** [comment]
It happened again, HmIP actuators are not controllable.

- [History.csv](https://github.com/danielperna84/hahomematic/files/12791394/History.csv)
- [home-assistant_2023-10-03T10-05-54.059Z.log](https://github.com/danielperna84/hahomematic/files/12791423/home-assistant_2023-10-03T10-05-54.059Z.log)

**Comment by ce3a** [comment]
If I try to close the shutters (HmIP-BROLL) or open the garage door (HmIP-MOD-HO), nothing happens. Also, I don't see anything related in the logs.

Can you help?

**Comment by ce3a** [comment]
Also, I cannot disable Bidcos-RF. If I open the integration configuration, Bidcos-RF is always enabled, even if I had previously disabled it.

**Comment by ce3a** [comment]
If I click on "reload" integration, nothing happens.

**Comment by ce3a** [comment]
I have enabled debug level logging. Now if I try to close the shutters, I can see the following in the log. However, nothing happens.

```
2023-10-03 15:02:30.934 DEBUG (MainThread) [hahomematic.client] SET_VALUE: 00111F299F10BB:4, LEVEL, 0.0
2023-10-03 15:02:30.935 DEBUG (MainThread) [hahomematic.client.xml_rpc] __ASYNC_REQUEST: ('setValue', ('00111F299F10BB:4', 'LEVEL', 0.0))
```

**Comment by SukramJ** [comment]
Looks like expected.

Anything in the CCU logs?

**Comment by ce3a** [comment]
Here the CCU log around the same time stamp.

```
14:58:33 | 03.10.2023 | RPI-RF-MOD 001F5F298D833A:0 | Duty Cycle 14.00%, Carrier Sense 0.00%
15:03:33 | 03.10.2023 | RPI-RF-MOD 001F5F298D833A:0 | Duty Cycle 14.00%, Carrier Sense 0.50%
15:08:33 | 03.10.2023 | RPI-RF-MOD 001F5F298D833A:0 | Duty Cycle 14.00%, Carrier Sense 0.50%
15:13:33 | 03.10.2023 | RPI-RF-MOD 001F5F298D833A:0 | Duty Cycle 13.00%, Carrier Sense 0.00%
15:18:33 | 03.10.2023 | RPI-RF-MOD 001F5F298D833A:0 | Duty Cycle 12.50%, Carrier Sense 0.00%
15:23:33 | 03.10.2023 | RPI-RF-MOD 001F5F298D833A:0 | Duty Cycle 13.00%, Carrier Sense 0.00%
```

**Comment by SukramJ** [comment]
I don't know anything I can do.

**Comment by ce3a** [comment]
I good case, after `__ASYNC_REQUEST` I can see something like the following the log:

```

DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = CONFIG_PENDING, value = False
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = RSSI_DEVICE, value = -80
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = DUTY_CYCLE, value = False
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = UNREACH, value = False
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = RSSI_PEER, value = -82
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:4, parameter = SECTION_STATUS, value = 0
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:4, parameter = LEVEL, value = 0.0
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:4, parameter = LEVEL_STATUS, value = 0
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:4, parameter = SECTION, value = 3
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:4, parameter = PROCESS, value = 1
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:4, parameter = ACTIVITY_STATE, value = 1
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = CONFIG_PENDING, value = False
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = RSSI_DEVICE, value = -81
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = DUTY_CYCLE, value = False
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:0, parameter = UNREACH, value = False
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:3, parameter = SECTION_STATUS, value = 1
DEBUG (XmlRpcServer on port 50859) [hahomematic.central] EVENT: interface_id = RaspberryMatic-HmIP-RF, channel_address = 00111F299F10BB:3, parameter = LEVEL, value = 0.895
```

In bad case, I dont see it.

**Comment by SukramJ** [comment]
The problem is not the integration.
The commands are send without a problem, but for some reasons, the CCU is not sending events with changes.
So I would locate the problem on the CCU.
As long there is nothing in any log, it's hard to identify the root cause.

**Comment by ce3a** [comment]
> The problem is not the integration.
> The commands are send without a problem, but for some reasons, the CCU is not sending events with changes.

But why does restarting ha resolve the issue then?

Not only the CCU is not sending events. There is also no action triggered, I mean the shutters do not move, for example.

**Comment by SukramJ** [comment]
A restart triggers a new connection init, that tells the ccu to start a connection to HA, that should work for hours, weeks and month. This is the same mechanism that is triggered when the integration is reloaded.
The minimum traffic between HA is a steady stream of PING and PONG events. Problems with PING/PONG are also reported as warnings/errors in the logs.

I can't tell you why there is a problem on your environment and why your CCU stops sending device events to HA, as long as there are no logs that provide these information.

What i know so far is that there is currently no other user reporting problems like you.

**Reply by ce3a** [reply]
If I trigger closing the shutters from the CCU WebUI, I can see the status being updated in HA. This tells me the connection from CCU to HA is ok. However, the connection from HA to CCU seems to be broken. It seems requests from HA do to reach the CCU.
